### ISSUE TITLE
Mitsue-Links Accessibility Dept. Comment on SC 2.4.12 Accessible Name: Visible non-text labels

### ISSUE THREAD

**Issue by honoka-h** [w3c/wcag21#504]
There are cases where visible labels can be non-text (especially images) alone (Even [WAI's Web Accessibility Tutorials has an example](https://www.w3.org/WAI/tutorials/images/functional/#stand-alone-icon-image-that-has-a-function).)

It is not clear whether non-text labels which have alternative texts conforming to SC 1.1.1 could conform to this SC or not.

Example 1: Combining text and image.

It is clear that the following example passes the SC.

```html
<a href="javascript:print()"><img src="print.png" alt="">Print this page</a>
```

Example 2: Stand-alone image

However, it is not clear if the following example would pass the SC or not.

```html
<a href="javascript:print()"><img src="print.png" alt="Print this page"></a>
```

Note: Above code is copied from https://www.w3.org/WAI/tutorials/images/functional/#stand-alone-icon-image-that-has-a-function

Please clarify how non-text labels should be handled in this SC.

**Comment by DavidMacDonald** [comment]
If there is no text in the label, then there is no requirement to include that in the label. The problem arises when the Dragon user says the words in the label but they are not in the name, which is what is triggered. In this case the Dragon User would only see a print icon so they wouldn't generally say "click print" because those words are nowhere to be found.

https://github.com/w3c/wcag21/issues/371

**Comment by steverep** [comment]
@honoka-h, I think you are correct that the language was not clear when it came to labels that did not include or only partially included text.  This was brought up in the context of a larger issue #371, and has now been resolved by the merge of #377.  The new version of the SC is called "Label in Name" and reads:

> **Label in Name:** For active user interface components with labels that include text, the name includes the text of the label.
>
> *NOTE:* A best practice is to have the text of the label at the start of the name.

This latest version can be viewed at http://rawgit.com/w3c/wcag21/master/guidelines/#label-in-name.

So, now the criterion is more clearly scoped to "labels that include text".  Does this resolve the issue?

**Comment by honoka-h** [comment]
@steverep 
Thank you for the suggestion.
We believe that the new version of SC "Label in Name" is fine, as long as the details are clearly explained in other WCAG supporting documents.

**Comment by joshueoconnor** [comment]
This has been dealt with in the merge  #377. The new version of the SC is called "Label in Name". Thanks @steverep and @honoka-h for the comment. This can be closed.
